algorithms.py

```diff
@@ -482,20 +576,123 @@ def hudson_simulate(self):
                 self.t += min_time
                 if min_time == t_re:
                     # print("RE EVENT")
-                    self.hudson_recombination_event()
+                    self.hudson_recombination_event(0)
                 elif min_time == t_ca:
                     # print("CA EVENT")
-                    self.common_ancestor_event(ca_population)
+                    self.common_ancestor_event(ca_population, 0)
                 else:
                     # print("MIG EVENT")
                     self.migration_event(mig_source, mig_dest)
         return self.finalise()
 
+    def single_sweep_simulate(self):
+        """
+        Does a structed coalescent until end_freq is reached, using
+        information in self.weep_trajectory.
+
+        """
+        allele_freqs, times = self.sweep_trajectory
+        sweep_traj_step = 0
+        x = allele_freqs[sweep_traj_step]
+
+        assert self.num_populations == 1
+
+        # go through segments and assign labels
+        # a bit ugly with the two loops because
+        # of dealing with the pops
+        indices = []
```

Comment:
We must be missing something here I think. The ``indices`` array is always empty, so the second loop never runs.

Reply:
this was vestigal... forgot to remove it from some checks i was performing

Reply:
I got a message on email where you had replied to this I think, but it's not showing up on github. Can I just delete these two loops then?

Reply:
no we need these-- the first loop goes through and switches the labels of segments. the second loop pops them out and moves them over. could probably condense into one loop but i wasn't sure of a clean pythonic way to do this, so left it as two loops for clarity